src/app/service-providers/utils/service-provider-filters.ts

```diff
@@ -0,0 +1,83 @@
+import type { FilterFn } from '@tanstack/react-table'
+
+import type { ServiceProvider } from '@/schemas/provider-schema'
+
+import { getYesNoFromBoolean } from './get-yes-no-from-boolean'
+import type { FilterState } from '../hooks/useFilterQueryState'
+
+type RangeFilter = {
+  min: FilterState['capacityMin']
+  max: FilterState['capacityMax']
+}
+
+export const statusFilterFn: FilterFn<ServiceProvider> = (
+  row,
+  _columnId,
+  filterValue,
+) => {
+  const statusArray = filterValue as FilterState['status']
+  if (statusArray.length === 0) return true
+
+  const serviceStatus = row.original.serviceStatus
+  return serviceStatus ? statusArray.includes(serviceStatus) : false
+}
+
+export const countryFilterFn: FilterFn<ServiceProvider> = (
+  row,
+  _columnId,
+  filterValue,
+) => {
+  const countryArray = filterValue as FilterState['country']
+  if (countryArray.length === 0) return true
+
+  const location = row.original.location
+  return countryArray.includes(location)
+}
+
+export const ipniFilterFn: FilterFn<ServiceProvider> = (
+  row,
+  _columnId,
+  filterValue,
+) => {
+  const ipniArray = filterValue as FilterState['ipni']
+  if (ipniArray.length === 0) return true
+
+  const ipniIpfs = row.original.ipniIpfs
+  const ipniValue = getYesNoFromBoolean(ipniIpfs)
+  return ipniArray.includes(ipniValue)
+}
+
+export const capacityRangeFilterFn: FilterFn<ServiceProvider> = (
+  row,
+  _columnId,
+  filterValue,
+) => {
+  const range = filterValue as RangeFilter
+  const capacity = row.original.capacityTb
+
+  if (!capacity) return false
+
+  const capacityNum = Number(capacity)
+  const { min, max } = range
```

Comment:
Original code excludes rows with `null` `capacityTb` even when no filter is active. Since `capacityTb` is optional in the schema, this could hide service providers by default.

Check if filter is applied first (both min/max are null) → show all rows. Only exclude null capacity when user actually sets a filter. Same for `provingPeriodRangeFilterFn`

```suggestion
  const { min, max } = range
  
  // If no filter is applied, show all rows including those without capacity data
  if (min === null && max === null) return true
  
  const capacity = row.original.capacityTb
  
  // If filter IS applied but row has no capacity data, exclude it from results
  if (!capacity) return false
  
  const capacityNum = Number(capacity)
```